test/libsyncengine/propagation/executor/testexecutorworker.cpp

```diff
@@ -178,32 +178,126 @@ SyncOpPtr TestExecutorWorker::generateSyncOperation(const DbNodeId dbNodeId, con
+    {
+        SyncOpPtr op = generateSyncOperationWithNestedNodes(1, Str("test_file.txt"), OperationType::Create, NodeType::File);
+        executorWorkerMock->setCorrespondingNodeInOtherTree({{op->affectedNode()->parentNode(), nullptr}});
+        op->setTargetSide(ReplicaSide::Remote);
+        CPPUNIT_ASSERT(!executorWorkerMock->isValidDestination(op));
     }
-    // Always true if the item is created on the local replica, at the root of the synchronisation folder
+
+    const auto root = _syncPal->updateTree(ReplicaSide::Remote)->rootNode();
+
+    // False if the item is created on the local replica is not at the root of the synchronisation folder and has a
+    // corresponding parent node with no id.
+    {
+        const auto correspondingParentNode = std::make_shared<Node>(
+                666, ReplicaSide::Remote, Str("parent_dir"), NodeType::Directory, OperationType::None, std::nullopt,
+                testhelpers::defaultTime, testhelpers::defaultTime, testhelpers::defaultFileSize, root);
+
+
+        SyncOpPtr op = generateSyncOperationWithNestedNodes(1, Str("test_file.txt"), OperationType::Create, NodeType::File);
+        executorWorkerMock->setCorrespondingNodeInOtherTree({{op->affectedNode()->parentNode(), correspondingParentNode}});
+        op->setTargetSide(ReplicaSide::Remote);
+        CPPUNIT_ASSERT(!executorWorkerMock->isValidDestination(op));
+    }
+
+    const auto correspondingParentCommonDocsNode = std::make_shared<Node>(
+            666, ReplicaSide::Remote, Utility::commonDocumentsFolderName(), NodeType::Directory, OperationType::None,
+            "common_docs_id", testhelpers::defaultTime, testhelpers::defaultTime, testhelpers::defaultFileSize, root);
+
+    // False if the item is created on the local replica is a file and has Common Documents as corresponding parent node.
+    {
+        SyncOpPtr op = generateSyncOperationWithNestedNodes(1, Str("test_file.txt"), OperationType::Create, NodeType::File);
+        op->setTargetSide(ReplicaSide::Remote);
+        executorWorkerMock->setCorrespondingNodeInOtherTree(
+                {{op->affectedNode()->parentNode(), correspondingParentCommonDocsNode}});
+        CPPUNIT_ASSERT(!executorWorkerMock->isValidDestination(op));
+    }
+
+    // True if the item is created on the local replica is a directory and has Common Documents as corresponding parent node.
     {
-        SyncOpPtr op = generateSyncOperation(1, Str("test_file.txt"), OperationType::Create);
+        SyncOpPtr op = generateSyncOperationWithNestedNodes(1, Str("test_dir"), OperationType::Create, NodeType::Directory);
         op->setTargetSide(ReplicaSide::Remote);
-        CPPUNIT_ASSERT(_syncPal->_executorWorker->isValidDestination(op));
+        executorWorkerMock->setCorrespondingNodeInOtherTree(
+                {{op->affectedNode()->parentNode(), correspondingParentCommonDocsNode}});
+        CPPUNIT_ASSERT(executorWorkerMock->isValidDestination(op));
     }
```

Comment:
Perhaps also add a test about the "Shared" folder. In this folder, it is forbidden to create files and directories

Reply:
Done with the last commit.